### ISSUE TITLE
Add more examples to explain Dereferencing

### ISSUE THREAD

**Issue by sb1752** [sb1752/bitcoiner-intro-to-rust#21]
Perhaps there can be more quiz questions or links to Rust Playground. Possibly another section 9.2.

**Comment by edilmedeiros** [comment]
What feedback or observations from office hours do you have regarding that?

**Comment by sb1752** [comment]
Yes, this was just one person that mentioned it in the weekly feedback form and another during office hours. I wanted to make sure this wasn't too confusing since I mentioned it so briefly in the course. 

Maybe the quiz does a decent job of getting that idea across? Maybe there's a good additional reading material I can add to the end. 

I also had an idea for something in Rust playground in which we take two structs (`Foo` and `Bar`) and implement the `Deref` trait on one of them to showcase how this works. Maybe I can add that as an example to follow at the end of the section, but we haven't introduced Structs yet.

This probably doesn't warrant a section 9.2 though now that I think about it, that might be too much.
